plugins/gimp/file-jxl-save.cc

```diff
@@ -3,130 +3,813 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+// #include "plugins/gimp/file-jxl-save.h"
+
 #include "plugins/gimp/file-jxl-save.h"
 
-// Defined by both FUIF and glib.
-#undef MAX
-#undef MIN
-#undef CLAMP
+#define PLUG_IN_BINARY "file-jxl"
+#define SAVE_PROC "file-jxl-save"
 
-#include "lib/jxl/alpha.h"
-#include "lib/jxl/base/file_io.h"
-#include "lib/jxl/base/thread_pool_internal.h"
-#include "lib/jxl/enc_file.h"
-#include "plugins/gimp/common.h"
+#define SCALE_WIDTH 200
 
 namespace jxl {
 
-namespace {
-
-template <bool has_alpha, size_t alpha_bits = 16>
-Status ReadBuffer(const size_t xsize, const size_t ysize,
-                  const std::vector<float>& pixel_data, PaddedBytes icc,
-                  CodecInOut* const io) {
-  constexpr float alpha_multiplier = has_alpha ? 1.f / 255 : 0.f;
-  Image3F image(xsize, ysize);
-  ImageF alpha;
-  if (has_alpha) {
-    alpha = ImageF(xsize, ysize);
-  }
-  const float* current_sample = pixel_data.data();
-  for (size_t y = 0; y < ysize; ++y) {
-    float* rows[3];
-    for (size_t c = 0; c < 3; ++c) {
-      rows[c] = image.PlaneRow(c, y);
+JpegXlSaveOpts jxl_save_opts;
+
+bool SaveJpegXlImage(const gint32 image_id, const gint32 drawable_id,
+                     const gint32 orig_image_id, const gchar* const filename) {
+  if (!save_dialog()) {
+    return true;
+  }
+
+  JpegXlGimpProgress gimp_save_progress(
+      ("Saving JPEG XL file:" + (std::string)filename).c_str());
```

Comment:
C++-style cast would be preferred:

```suggestion
      ("Saving JPEG XL file:" + std::string(filename)).c_str());
```

Reply:
Thanks.  Will fix this.